examples/wavenet/modules.py

```diff
@@ -0,0 +1,83 @@
+import chainer
+import chainer.functions as F
+import chainer.links as L
+
+
+class ResidualBlock(chainer.Chain):
+    def __init__(self, filter_size, dilation,
+                 residual_channels, dilated_channels, skip_channels):
+        super(ResidualBlock, self).__init__()
+        with self.init_scope():
+            self.conv = L.DilatedConvolution2D(
+                residual_channels, dilated_channels,
+                ksize=(filter_size, 1),
+                pad=(dilation * (filter_size - 1), 0), dilate=(dilation, 1))
+            self.res = L.Convolution2D(
+                dilated_channels // 2, residual_channels, 1)
+            self.skip = L.Convolution2D(
+                dilated_channels // 2, skip_channels, 1)
+
+        self.filter_size = filter_size
+        self.dilation = dilation
+        self.residual_channels = residual_channels
+
+    def __call__(self, x, condition):
+        length = x.shape[2]
+        h = self.conv(x)
```

Comment:
I just noticed that a single conv layer is applied before different activation functions in this implementation, while [the original paper](https://arxiv.org/pdf/1609.03499.pdf) looks like it uses different conv layers (W_{f,k} and W_{g,k} in Eq.(2)).

Reply:
Two convolutions can be represented by single convolution and splitting in channel axis after convolution.

https://github.com/dhgrs/chainer/blob/48f7786371bf0b0c40a4504426ef4e18bc34e888/examples/wavenet/modules.py#L29

In figure 4 in the original paper looks like single convolution and splitting are used. And nv-wavenet, CUDA kernel for WaveNet inference also uses single convolution and splitting instead of two convolutions.

Reply:
I see. Thank you for your response!